blueocean-rest/src/main/java/io/jenkins/blueocean/rest/hal/Link.java

```diff
@@ -28,4 +29,27 @@ public String getHref(){
     public Link rel(String name) {
         return new Link(href+name);
     }
+
+
+    /**
+     * Gives ancestor Link, for example, for link("/a/b/c/d"), ancestor is "a/b/c/".
+     */
+    public Link ancestor(){
+        int i = href.lastIndexOf("/");
+
+        if(i>0) {
+            int j = href.substring(0, i).lastIndexOf("/");
```

Comment:
hm. `assert` doesn't typically execute at runtime. but this sounds good, I assume most of the cases are covered by tests. 👍